protocol/wallet.rei

```diff
@@ -1,14 +1,26 @@
 open Mirage_crypto_ec;
-open Helpers;
 
 [@deriving (ord, yojson)]
 type t;
+[@deriving (ord, yojson)]
+type pub_;
+
+let genesis_key: t;
+let genesis_pubkey: pub_;
+
+let wallet_of_privkey: Ed25519.priv => t;
+let wallet_to_privkey: t => Ed25519.priv;
+let pub_of_Ed25519pub: Ed25519.pub_ => pub_;
+let pub_to_Ed25519pub: pub_ => Ed25519.pub_;
+
+let pubkey_of_wallet: t => pub_;
+let make_pair: unit => (t, pub_);
+let make_pubkey: unit => pub_;
+
+let to_hex: string => string;
+let of_hex: string => string;
```

Comment:
Pretty sure of_hex cannot have this type and should not be exported here, currently we avoid exception as we don't have an effect system.

Reply:
What type does it have here? https://github.com/marigold-dev/sidechain/blob/main/protocol/address.re#L10 my editor says `string => string`. Probably right that it shouldn't be exported

Reply:
Yes, but we handle the exception internally

at > https://github.com/marigold-dev/sidechain/blob/main/protocol/address.re#L18